[Table] Rename onChangeRowsPerPage, onChangePage (#22900)
mui-org/material-ui: docs/src/pages/guides/api/api.md

Most of the controlled component are controlled via the `value` and the `onChange` props,
however, the `open` / `onClose` / `onOpen` combination is used for display related state.

> As a design approach, we decided to put the noun first, and then the verb. for example - `onPageChange`

Reviewer: ```suggestion
In the cases where there are more events, we put the noun first, and then the verb, for example: `onPageChange`, `onRowsChange`.
```
